core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/UserApiService.scala

```diff
@@ -23,126 +23,147 @@ trait UserApiService extends HttpService with PerRequestCreator with UserInfoDir
+        detach() {
+          post {
+            requestContext =>
+              perRequest(requestContext,
+                UserService.props(userServiceConstructor, userInfo),
+                UserService.CreateUser)
+          }
+        }
       }
     }
   }
-
   // standard /api routes begin here
 
-  val userRoutes = requireUserInfo() { userInfo =>
-    path("user" / "refreshToken") {
-      put {
-        entity(as[UserRefreshToken]) { token =>
-          requestContext => perRequest(requestContext,
-            UserService.props(userServiceConstructor, userInfo),
-            UserService.SetRefreshToken(token))
-        }
-      }
-    } ~
-    path("user" / "refreshTokenDate") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.GetRefreshTokenDate)
-      }
-    } ~
-    path("user" / "billing") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.ListBillingProjects)
-      }
-    } ~
-    path("user" / "role" / "admin") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.IsAdmin(userInfo.userEmail))
-      }
-    } ~
-    path("user" / "role" / "curator") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.IsLibraryCurator(userInfo.userEmail))
-      }
-    } ~
-    path("user" / "billingAccounts") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.ListBillingAccounts)
-      }
-    } ~
-    path("user" / "groups") {
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.ListGroupsForUser(userInfo.userEmail))
-      }
-    } ~
-    path("user" / "group" / Segment) { groupName =>
-      get {
-        requestContext => perRequest(requestContext,
-          UserService.props(userServiceConstructor, userInfo),
-          UserService.GetUserGroup(RawlsGroupRef(RawlsGroupName(groupName))))
-      }
-    } ~
-    pathPrefix("groups") {
-      pathEnd {
-        get {
-          requestContext => perRequest(requestContext,
-            UserService.props(userServiceConstructor, userInfo),
-            UserService.ListManagedGroupsForUser)
+  val userRoutes = detach() {
+    requireUserInfo() { userInfo =>
+      path("user" / "refreshToken") {
+        put {
+          entity(as[UserRefreshToken]) { token =>
+            requestContext =>
+              perRequest(requestContext,
+                UserService.props(userServiceConstructor, userInfo),
+                UserService.SetRefreshToken(token))
+          }
```

Comment:
note to other reviewers: this giant blob of diff for the rest of this file is because GitHub gets confused by whitespace changes. There's no code changes here; appending ?w=1 to the end of your URL will confirm.